### PR TITLE
perf(jqLite): avoid repeated add/removeAttribute in jqLiteRemoveClass

### PR DESCRIPTION
**Pull request by jbedard**
This does something similar to what was suggested in #16078, although keeps it a little more like `jqLiteAddClass` for now.

### CODE REVIEW

**Review by gkalpak — APPROVED** [review]
LGTM (as soon as Travis is happy)

File: src/jqLite.js [hunk 1]
```diff
@@ -420,13 +420,15 @@ function jqLiteHasClass(element, selector) {
...
-      element.setAttribute('class', trim(
-          (' ' + (element.getAttribute('class') || '') + ' ')
-          .replace(/[\n\t]/g, ' ')
-          .replace(' ' + trim(cssClass) + ' ', ' '))
-      );
+      cssClass = trim(cssClass);
+      existingClasses = existingClasses.replace(' ' + cssClass + ' ', ' ');
     });
+
+    element.setAttribute('class', trim(existingClasses));
```
Comment (dcherman) [inline src/jqLite.js]: Should you do a check to ensure that we don't set this unnecessarily which potentially causes unnecessary renders?

Ref: https://github.com/jquery/jquery/blob/master/src/attributes/classes.js#L41-L45

Comment (jbedard) [inline src/jqLite.js]: I thought that should be done along with addClass in its own commit...

**Comment by jbedard** [comment]
Passed now. However I decided to add one more commit like @dcherman suggested to only actually set the `class` attribute if it changed (for both `addClass` and `removeClass`).

**Review by gkalpak — APPROVED** [review]
LGTM
Do you think it is worth adding some tests to verify the new behavior (i.e. that we don't call `get/setAttribute()` more times than necessary)?

**Comment by Narretz** [comment]
@jbedard can you please rebase this? The tests currently don't run because of a Travis problem at the branch point.

**Comment by jbedard** [comment]
Rebased.

The tests had to be put in `if (_jqLiteMode)` since they are testing a JQLite implementation detail (I think jquery3 tests passed, but not jquery2).

**Review by mgol — APPROVED** [review]

File: test/jqLiteSpec.js [hunk 2]
```diff
@@ -914,6 +914,37 @@ describe('jqLite', function() {
       });
 
 
+      //JQLite specific implementation/performance tests
```
Comment (mgol) [inline test/jqLiteSpec.js]: A space before `JQLite`?

File: test/jqLiteSpec.js [hunk 3]
```diff
@@ -1031,6 +1062,37 @@ describe('jqLite', function() {
         jqA.removeClass('foo baz noexistent');
         expect(a.className).toBe('bar');
       });
+
+
+      //JQLite specific implementation/performance tests
```
Comment (mgol) [inline test/jqLiteSpec.js]: A space before `JQLite`?

File: test/jqLiteSpec.js (hunk 2 above)
Comment (jbedard) [inline test/jqLiteSpec.js]: NOTE: add space
